presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -192,8 +197,10 @@ public synchronized void addLocation(URI location, TaskId remoteSourceTaskId)
         if (location.getScheme().toLowerCase(Locale.ENGLISH).startsWith("http")) {
             resultClient = new HttpTaskResultClient(httpClient, location);
         }
+        else if (location.getScheme().toLowerCase(Locale.ENGLISH).startsWith("thrift")) {
+            resultClient = new ThriftTaskResultClient(driftClient, location, pageBufferClientCallbackExecutor);
```

Comment:
This `pageBufferClientCallbackExecutor` is used by `ThriftTaskResultClient#acknowledgeResults`. Curious why `HttpTaskRresultClient` doesn't need an executor for acknowledge result? Also shall we have dedicator executor for this (instead of reusing the `pageBufferClientCallbackExecutor`? )

Reply:
`HttpClient` has its own executor in `JettyHttpClient` and can be passed into jetty lib; execAsync may just reuse that executor. Drift also has its own executor but not exposed to the users. I'm thinking having a light-weighted executor with a limited request queue. If ack sends out, it's good; otherwise, it's fine too. Do you have any good executor candidate?